<a href="https://colab.research.google.com/github/Slalani555/Deep-Learning-Python/blob/main/Linear_Relationship_model.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### **Basic example of Machine Learning using Keras**

- Problem Statement- Predict y value for x value (input value)

*The neural network architechture*

In [1]:
from tensorflow import keras
from tensorflow.keras import layers
model=keras.Sequential([layers.Dense(units=1,input_shape=[1])])

*The compilation step*

In [2]:
model.compile(optimizer='sgd',loss='mean_squared_error')

In [8]:
import numpy as np
x=np.array([-3.0,-2.0,-1.0,0.0,1.0,2.0,3.0],dtype=float)
y=np.array([ -17.0,-12.0,-7.0,-2.0,3.0,8.0,12.0],dtype=float)

*Fitting the model*

In [9]:
model.fit(x,y,epochs=500)

Epoch 1/500
1/1 [==============================] - 0s 9ms/step - loss: 34.9327
Epoch 2/500
1/1 [==============================] - 0s 7ms/step - loss: 29.7298
Epoch 3/500
1/1 [==============================] - 0s 7ms/step - loss: 25.3200
Epoch 4/500
1/1 [==============================] - 0s 5ms/step - loss: 21.5818
Epoch 5/500
1/1 [==============================] - 0s 7ms/step - loss: 18.4123
Epoch 6/500
1/1 [==============================] - 0s 8ms/step - loss: 15.7243
Epoch 7/500
1/1 [==============================] - 0s 9ms/step - loss: 13.4441
Epoch 8/500
1/1 [==============================] - 0s 5ms/step - loss: 11.5093
Epoch 9/500
1/1 [==============================] - 0s 5ms/step - loss: 9.8670
Epoch 10/500
1/1 [==============================] - 0s 5ms/step - loss: 8.4724
Epoch 11/500
1/1 [==============================] - 0s 9ms/step - loss: 7.2877
Epoch 12/500
1/1 [==============================] - 0s 5ms/step - loss: 6.2808
Epoch 13/500
1/1 [==============================] - 0

In [10]:
print(model.predict([8.0]))

[[37.000027]]


Observation:-

The answers we fed to train our model had the rule  y=5x-2 and based on the rule, the answer should have been 38. Here, we predict our result to be 37.000027 which is close to 38 but not the exact value, since we fed only 7 numbers to our network which assumes a very high probability of having a linear relationship between x's and y's but we are not certain to have a linear relationship outside of the range provided to the network.  